Sources/swiftML/Helpers.swift

```diff
@@ -0,0 +1,70 @@
+import TensorFlow
+
+extension Tensor where Scalar: TensorFlowFloatingPoint {
+    /// The (Moore-Penrose) pseudoinverse of a matrix using its singular-value decomposition (SVD).
+    public var pseudoinverse: Tensor {
+        let svd = Raw.svd(self)
+        let diag = Raw.diag(diagonal: svd.s)
+        return matmul(matmul(svd.v, Raw.matrixInverse(diag)), svd.u.transposed())
+    }
+}
+
+/// Returns sign-corrected versions of the left and right singular vectors from
+/// single value decomposition (SVD).
+///
+/// - Parameters:
+///   - u: The tensor containing the left singular vectors for each matrix.
+///   - v: The tensor containing the right singular vectors for each matrix.
+/// - Returns: The sign corrected versions of `u` and `v` to ensure
+///   deterministic output.
+public func svdFlip<Scalar: TensorFlowFloatingPoint>(
```

Comment:
It seems that `svdFlip` takes the output of [`TensorFlow.Raw.Svd`](https://gist.github.com/dan-zheng/f85dfccdb157fed9ea1e82a98ce5a3ea) and returns a sign corrected version.

```swift
// Simplified type:
svdFlip: (Tensor, Tensor, uBasedDecision: Bool) -> (Tensor, Tensor)
```

Instead, how about creating `func signCorrectedSvd` which takes the same inputs as `TensorFlow.Raw.Svd`, calls `TensorFlow.Raw.Svd`, then performs sign correction at the end?
```
public func signCorrectedSvd<T: FloatingPoint & TensorFlowScalar>(
    _ input: Tensor<T>,
    fullMatrices: Bool = false, // perhaps this flag should be removed?
    uBasedDecision: Bool = true // rename this flag and add doc comment?
) -> (s: Tensor<T>, u: Tensor<T>, v: Tensor<T>) {
  var (s, u, v) = Raw.svd(input, computeUv: true, fullMatrices: fullMatrices)
  // Copy current sign correction logic.
  return (s, u, v)
}
```

Reply:
`svdFlip` used to flip eigenvectors' sign to enforce deterministic output for PCA algorithm. How about renaming it to `deterministicSvd` or 'determinantSvd`?
 
uBasedDecision:  Represent the variable on which the decision of svdFlip dependents.
If True, use the columns of `u` as the basis for sign flipping. Otherwise, use the rows of `v`.  

Right now, only columns of `u` used as the basis for sign flipping but row `v` based keep for the future.

[reference](https://github.com/scikit-learn/scikit-learn/blob/53f76d1a24ef42eb8c620fc1116c53db11dd07d9/sklearn/utils/extmath.py#L482)